modules/dnn_superres/include/opencv2/dnn_superres_quality.hpp

```diff
@@ -0,0 +1,116 @@
+// This file is part of OpenCV project.
+// It is subject to the license terms in the LICENSE file found in the top-level directory
+// of this distribution and at http://opencv.org/license.html.
+
+#ifndef OPENCV_CONTRIB_DNN_SUPERRES_QUALITY_HPP
+#define OPENCV_CONTRIB_DNN_SUPERRES_QUALITY_HPP
+
+#include <opencv2/highgui.hpp>
+#include <opencv2/imgproc.hpp>
+#include <cstdarg>
+#include <sstream>
+#include <ctime>
+
+#ifdef _WIN32
+#include <Windows.h>
+#endif
```

Comment:
Do we really need platform specific include in public includes?

> #include <opencv2/highgui.hpp>
> ...

A lot of these headers are not needed to define this API.
Move these includes into corresponding .cpp files.

Reply:
Corrected. I also don't need platform specific include anymore, and eliminated it.